test/extensions/filters/http/local_ratelimit/filter_test.cc

```diff
@@ -45,6 +45,39 @@ stat_prefix: test
       key: x-local-ratelimited
       value: 'true'
 local_rate_limit_per_downstream_connection: {}
+enable_x_ratelimit_headers: {}
+  )";
+static const std::string config_yaml_resource_exhausted = R"(
+stat_prefix: test
+rate_limited_as_resource_exhausted: true
```

Comment:
Could we just add a new `rate_limited_as_resource_exhausted: {}` in the original `config_yaml` and use it by setting parameter of `fmt::format` to true